genesis/engine/entities/rigid_entity/rigid_link.py

```diff
@@ -121,82 +122,88 @@ def _build(self):
         for vgeom in self._vgeoms:
             vgeom._build()
 
-        self._init_mesh = self._compose_init_mesh()
-
-        # inertial_mass and inertia_i
-        if self._inertial_mass is None:
-            if len(self._geoms) == 0 and len(self._vgeoms) == 0:
-                self._inertial_mass = 0.0
+        if self._inertial_mass is None or self._inertial_pos is None or self._inertial_i is None:
+            # Determine which geom list to use: geoms first, then vgeoms, then fallback
+            if len(self._geoms) > 0:
+                geom_list = self._geoms
+                get_mesh = lambda geom: trimesh.Trimesh(geom.init_verts, geom.init_faces)
+                get_pos = lambda geom: geom._init_pos
+                get_quat = lambda geom: geom._init_quat
+            elif len(self._vgeoms) > 0:
+                geom_list = self._vgeoms
+                get_mesh = lambda geom: trimesh.Trimesh(geom.init_vverts, geom.init_vfaces)
+                get_pos = lambda geom: geom._init_pos
+                get_quat = lambda geom: geom._init_quat
             else:
-                if self._init_mesh.is_watertight:
-                    self._inertial_mass = self._init_mesh.volume * self.entity.material.rho
-                else:  # TODO: handle non-watertight mesh
-                    self._inertial_mass = 1.0
+                # Fallback: use default values
+                self._inertial_mass = 1e-3
+                self._inertial_pos = np.zeros(3, dtype=gs.np_float)
+                self._inertial_i = np.zeros((3, 3), dtype=gs.np_float)
+                geom_list = []
+
+            # Process each geom individually and compose their properties
+            if len(geom_list) > 0:
+                total_mass = 0.0
+                total_com = np.zeros(3, dtype=gs.np_float)
+                total_inertia = np.zeros((3, 3), dtype=gs.np_float)
+
+                for geom in geom_list:
+                    inertia_mesh = get_mesh(geom)
+                    geom_pos = get_pos(geom)
+                    geom_quat = get_quat(geom)
+
+                    if inertia_mesh.is_watertight:
+                        geom_mass = inertia_mesh.volume * self.entity.material.rho
+                    else:
+                        geom_mass = 1e-3
+
+                    geom_com_local = np.array(inertia_mesh.center_mass, dtype=gs.np_float)
+
+                    if not inertia_mesh.is_watertight:
+                        inertia_mesh = trimesh.convex.convex_hull(inertia_mesh)
+
+                    if inertia_mesh.is_watertight and inertia_mesh.mass > 0:
```

Comment:
This condition is completely irrelevant. A convex hull cannot be non-watertight.